Added fileversion twig filter

jeanniton-mnr opened the pull request:
### fileversion
Forces the browser to reload any cached asset files that have been modified.

#### usage
```
<img src="{{ 'assets/images/image_file.jpg'|theme|fileversion}}" alt="an image>
```

Will return something like
```
<img src="https://www.example.com/themes/my-theme/assets/image_file.png?12.03.16.04.52.38" alt="an image">
```
See: https://github.com/vojtasvoboda/oc-twigextensions-plugin/issues/25

Commit:
Added fileversion twing filter

jeanniton-mnr committed:
Update README.md

vojtasvoboda commented:
I like the idea! But need to discuss a few points:

1. I'm thinking about filter name. Is the "fileversion" used somewhere else? Where does your inspiration came from? I'm thinking that maybe "revision" or just "version" will be better filter name.

2. I would add parameter name, let's say `?v=version`, to have correct URL format.

3. I'm not sure about version parameter format. Can we just use raw timestamp, to save some time on `date` function. I think `image.jpg?v=1454166258` is totally fine.

4. If file doesn't exists, I would return just filename, without `?`.

What do you think?

jeanniton-mnr commented:
1. I think you are correct for renaming the filter to something like "version"; it simpler.
2. ok
3. The date format I am using is: `date ("m.d.y.H.i.s", $timestamp)`. I like it because it's more human readable than just raw timestamp. Let me know if you still find raw timestamp better though.

   ### The parameters' description:
    m: _Numeric representation of a month, with leading zeros_
    d: _Day of the month, 2 digits with leading zeros_
    y: _A two digit representation of a year_
   H: _24-hour format of an hour with leading zeros_
    i: _Minutes with leading zeros_
   s: _Seconds, with leading zeros_

 4. ok

## Inspirations:
https://css-tricks.com/can-we-prevent-css-caching/
https://laravel.com/docs/5.3/elixir#versioning-and-cache-busting

NB: One advantage over elixir is that this will work for any asset file, not just CSS & Js.

vojtasvoboda commented:
3) Yes, it's more human readable, but I'm thinking about performance - timestamp would be definitely faster (consider some gallery with 60 images in a page). And the second point is the time format - for example in the Czech Republic we are using d.m.Y.H.i.s.

So how about putting format as the first parameter? And without parameter, it will be timestamp?

`non-existsing-image.jpg | version` - returns `non-existing-image.jpg`
`image.jpg | version` - returns `image.jpg?v=1454166258`
`image.jpg | version("m.d.y.H.i.s")` - returns `image.jpg?v=12.03.16.04.52.38`

Other points are fine for me.

jeanniton-mnr commented:
Yeah!
Let's just give the user the choice of using whether a timestamp or a date and its format.
I will put this motion now !!! Thanks

jeanniton-mnr committed:
Filter name and behaviour updated

. Renamed the filter name to "revision"
. Give the user the choice of using whether a timestamp or a date and its format

See our last conversation here:
https://github.com/vojtasvoboda/oc-twigextensions-plugin/pull/26  

Sorry for the delay

Thanks!!!

jeanniton-mnr committed:
Filter name and behaviour updated

jeanniton-mnr commented:
Hello @vojtasvoboda!

I have just done with the update. Sorry for getting back to this late.

vojtasvoboda commented:
Plugin version 1.1.1 ready at the Market. Thanks for contribution!
